app/src/main/java/unipd/se18/ocrcamera/ImageProcessing.java

```diff
@@ -0,0 +1,349 @@
+import java.io.FileOutputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.opencv.imgproc.Imgproc.INTER_CUBIC;
+import static org.opencv.imgproc.Imgproc.getRectSubPix;
+import static org.opencv.imgproc.Imgproc.getRotationMatrix2D;
+
+
+/**
+ * Class used to analyze the image
+ */
+public class ImageProcessing {
+
+    /*
+        Documentation of the Imgproc class available at:
+        https://docs.opencv.org/java/2.4.2/org/opencv/imgproc/Imgproc.html
+        
+        We referenced a previous instance of the documentation since
+        the newer one is still incomplete
+
+        We are trying to implement all the Imgproc methods,
+        which require a large number of
+        parameters to be called, with a builder statement,
+        to simplify reading and eventually changing them
+        (at the moment there are a little too many "magic numbers")
+     */
+
+    //Tag used to identify the log
+    final String TAG = "openCV";
+
+
+    /**
+     * Constructor of the class which initialize the openCV library
+     * @author Thomas Porro (g1)
+     */
+    public ImageProcessing() {
+        //TODO verify if the library is correctly loaded
+        //Load the openCV library
+        System.loadLibrary("opencv_java3");
+        Log.i(TAG, "Loaded the library");
+    }
+
+    /**
+     * Find, crop and rotate the text in an image
+     * @param imagePath the path of the image you want to analyze
+     * @return the cropped image
+     * @throws FileNotFoundException if imagePath doesn't exist
+     * @author Thomas Porro (g1)
+     */
+    public Bitmap findText(String imagePath) throws FileNotFoundException {
+        //Converts the image into a matrix
+        Mat img = Imgcodecs.imread(imagePath);
+
+        //Call of internal methods
+        RotatedRect area = detectMaxTextArea(imagePath);
+        img = crop(area, img);
+        Bitmap image = conversion(img);
+        return image;
+    }
+
+
+    /**
+     * Calculate the angle between the text and the horizontal
+     * @param imagePath path of the image you want to analyze
+     * @return the angle between the text and the horizontal
+     * @throws FileNotFoundException if imagePath doesn't exist
+     * @author Thomas Porro (g1)
+     */
+    public double computeSkew(String imagePath) throws FileNotFoundException {
+
+        //Turns the image in grayscale and put it in a matrix
+        Log.d(TAG, "Image path = " + imagePath);
+        Mat img = conversion(imagePath);
+        //save(img, "grayScale.jpg");
+
+        //Invert the colors of "img" onto itself
+        Core.bitwise_not(img, img);
+
+        //Detect the edges in the image
+        double threshold1 = 50;
+        double threshold2 = 200;
+        int apertureSize = 3;
+        boolean l2gradient = false;
+        Imgproc.Canny(img, img, threshold1, threshold2, apertureSize, l2gradient);
```

Comment:
I Don't get exactly what this method does

Reply:
It detects the edges of the image, in case of any doubt the documentation is linked at the beginning of the class